Name inference for unbind

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25582 Name inference for min(Tensor, dim?) / max(Tensor, dim?)
* **#25585 Name inference for unbind**

Test Plan:
- new tests [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25585

Differential Revision: [D17185070](https://our.internmc.facebook.com/intern/diff/D17185070)

zou3519 committed:
Name inference for argmax, argmin

Test Plan:
- new tests [namedtensor ci]

zou3519 committed:
Update on "Name inference for argmax, argmin"


Test Plan:
- new tests [namedtensor ci]

zou3519 committed:
Update on "Name inference for unbind"


Test Plan:
- new tests [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25585

zou3519 committed:
Update on "Name inference for unbind"


Test Plan:
- new tests [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25585

Differential Revision: [D17185070](https://our.internmc.facebook.com/intern/diff/D17185070)

nairbv commented on test/test_namedtensor.py:
```diff
@@ -741,26 +741,29 @@ def get_minmax_output(t):
+            Case('var_mean', True, True, False, True, None),
+            Case('min', True, False, True, True, get_minmax_output),
+            Case('max', True, False, True, True, get_minmax_output),
+            Case('unbind', False, False, False, False, None),
```
if all false, what are we testing for `unbind` ?

zou3519 replied:
There's a default set of tests that run if everything is false. That default set is {test_simple_reduce(op_name, device)} right now.

nairbv commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -1055,6 +1055,12 @@ std::vector<Tensor> unbind(const Tensor &self, int64_t dim) {
 
+#ifdef BUILD_NAMEDTENSOR
+std::vector<Tensor> unbind(const Tensor& self, Dimname dim) {
+  return at::unbind(self, dimname_to_position(self, dim));
```
I'd expect to see some change in the existing unbind method to remove the one name (if named) from each resulting unbound tensor. Is that not the expected behavior?

zou3519 replied:
`unbind` calls `select` which does that already.

zou3519 committed:
Update on "Name inference for unbind"


Test Plan:
- new tests [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25585

Differential Revision: [D17185070](https://our.internmc.facebook.com/intern/diff/D17185070)
